### PR TITLE
feat: multiple block can contain the same message hash

### PR DESCRIPTION
**Pull request by zengchen221**
### Proposed changes in this pull request
multiple block can contain the same message hash

### Type
- [ ] Bug fix: (Link to the issue #{issue No.})
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION](https://github.com/qlcchain/go-qlc/blob/master/CONTRIBUTING.md).
- [x] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable.
- [x] Code has been written according to [Golang-Style-Guide](https://github.com/qlcchain/go-qlc/blob/master/CONTRIBUTING.md#code-standard)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: rpc/api/util.go [hunk 1]
```diff
@@ -25,6 +29,19 @@ func (u *UtilApi) Encrypt(raw string, passphrase string) (string, error) {
 }
 
 func (u *UtilApi) RawToBalance(balance types.Balance, unit string, tokenName *string) (types.Balance, error) {
+	if tokenName != nil {
+		token, err := u.ledger.GetTokenByName(*tokenName)
+		if err != nil {
+			return types.ZeroBalance, err
+		}
+		if token.TokenId != common.ChainToken() {
+			b, err := balance.Div(int64(math.Pow10(int(token.Decimals))))
```
Comment (gythialy) [inline rpc/api/util.go]: should check `math.Pow10` result, it's could be zero.

Comment (zengchen221) [reply rpc/api/util.go]: token.Decimals is nonnegative, so zero should not happen.  and if zero happen ,Div will return error , no panic

Comment (gythialy) [reply rpc/api/util.go]: when `token.Decimals` is big than 308, the result is not a number.

Comment (gythialy) [reply rpc/api/util.go]: forget about it, `Decimals`'s type is uint8...

**Pull request merged**